generators/add-action/customer-profile/index.js

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const path = require('path')
+const ActionGenerator = require('../../../lib/ActionGenerator')
+
+class CustomerProfileGenerator extends ActionGenerator {
+  constructor (args, opts) {
+    super(args, opts)
+    this.props = {
+      description: 'This is a sample action showcasing how to access an external Customer Profile API',
+      // eslint-disable-next-line quotes
+      requiredParams: `['tenantId', 'iMSOrgId', 'apiKey']`,
+      // eslint-disable-next-line quotes
+      importCode: `const CustomerProfileSDK = require('@adobe/aio-lib-customer-profile')`,
+      responseCode: `// initialize sdk
+    const client = await CustomerProfileSDK.init(params.tenantId, params.iMSOrgId, params.apiKey, token)
+    // call methods, eg getSegmentRoute
+    const response = await client.getExperienceEvents({
```

Comment:
After setting my .env and deploying the app, I tried running the action in my generated template like:
`curl https://<myns>.adobeioruntime.net/api/v1/web/<myappname>-0.0.1/customer-profile  -H "Authorization: Bearer xxx"`
The action was failing and looking at the logs I got:
```
Error 400 - Bad Request ({"message":"UPS-ACC-1000002:Some parameters are not specified - \"relatedEntityId,entityId\".","statusCode":400,"recommendedAction":"Provide listed parameters."})
```
I might be missing something as I didn't use the customer profile sdk before, but we should make sure that the template action and the call to the sdk are working ootb

Reply:
In order to work OOTB, I can use only few SDK methods, since most of the methods require data to already exist, and to provide input params (like user email, etc). And this methods without required params are rarely used, so I would like to stick with Profile related methods. Is there any way to pass required method params to generator?

Reply:
Yes that's fine, you are right I missed the required parameters, just make sure to add the required parameters to the list as mentioned in my comment here https://github.com/adobe/generator-aio-app/pull/30#discussion_r405650393. The action would then return a 400  with `missing param xyz` instead of 500 `server error`